go/stellar/global.go

```diff
@@ -686,6 +686,26 @@ func (s *Stellar) OwnAccountCached(mctx libkb.MetaContext, accountID stellar1.Ac
 	return OwnAccount(mctx, accountID)
 }
 
+func (s *Stellar) OwnAccountPlusNameCached(mctx libkb.MetaContext, accountID stellar1.AccountID) (own, isPrimary bool, accountName string, err error) {
+	err = libkb.AcquireWithContextAndTimeout(mctx.Ctx(), &s.accountsLock, 5*time.Second)
+	if err != nil {
+		mctx.Debug("OwnAccountPlusNameCached: error acquiring lock")
+		return
+	}
+	if s.accounts != nil && mctx.G().Clock().Now().Sub(s.accounts.Stored.Round(0)) < 2*time.Minute {
+		for _, acc := range s.accounts.Accounts {
+			if acc.AccountID.Eq(accountID) {
+				s.accountsLock.Unlock()
+				return true, acc.IsPrimary, acc.Name, nil
+			}
+		}
+		s.accountsLock.Unlock()
+		return false, false, "", nil
+	}
+	s.accountsLock.Unlock()
+	return OwnAccountPlusName(mctx, accountID)
+}
+
```

Comment:
dedup by having `OwnAccountCached` call `OwnAccountPlusNameCached` and discard the name.